library/std/src/thread/mod.rs

```diff
@@ -446,6 +452,20 @@ impl Builder {
         F: FnOnce() -> T,
         F: Send + 'a,
         T: Send + 'a,
+    {
+        Ok(JoinHandle(unsafe { self.spawn_unchecked_(f, None) }?))
+    }
+
+    unsafe fn spawn_unchecked_<'a, 'scope, F, T>(
+        self,
+        f: F,
+        scope_data: Option<&'scope scoped::ScopeData>,
+    ) -> io::Result<JoinInner<'scope, T>>
+    where
+        F: FnOnce() -> T,
+        F: Send + 'a,
+        T: Send + 'a,
+        'scope: 'a,
```

Comment:
Now that we are at it, and to try and fight a bit the not-so-great habit of `'a`-named lifetimes: could these be named `'env`?

Reply:
It doesn't seem to be the same thing. `'scope` outlives `'a`, but `'env` outlives `'scope`.

Reply:
I personally like calling `F : 'region` as `F : 'env`, but you're right that it would be misleading with the outer `'env` 💯 so nevermind my comment.

Tangentially, however, could we remove the `'a` from the `pub` `spawn_unchecked` at least?

Reply:
> Tangentially, however, could we remove the `'a` from the `pub` `spawn_unchecked` at least?

That sounds like something for a separate PR. But just like `T` and `F` are the standard names for generic types and function types, having a standard name (`'a`) for a lifetime seems fine to me. But feel free to start that discussion on a separate issue or PR. :)